lib/index.js

```diff
@@ -1,8 +1,76 @@
 import fetch from 'node-fetch';
-import { promises as fs } from 'fs';
+import fs from 'fs';
 
-fetch('https://www.reddit.com/r/pics/search.json?q=programming&sort=new')
-  .then(res => res.json())
-  .then((res) => fs.writeFile('data.json', JSON.stringify(res)))
-  .then(() => console.log('Saved Successfully'))
-  .catch(err => console.error(err));
\ No newline at end of file
+// Set the fetch -> json into a helper function instead of repeating them
+
+export const searchRedditPostsByTopicPromise = (topic, fields = [], sort = "new") => {
+
+  const url = getRedditUrl(topic, sort);
+
+  fetchData(url)
+    .then(res => processData(res, fields))
+    .then(posts => saveResults('data.json', posts))
+    .catch(err => console.error('Error: ', err));
+}
+
+export const searchRedditPostsByTopicAsync = async (topic, fields = [], sort = "new") => {
+
+  const url = getRedditUrl(topic, sort);
+  
+  try { 
+    const data = await fetchData(url);
+    const posts = processData(data, fields);
+
+    saveResults('data.json', posts);
+  } catch(err) {
+    console.error(err);
+  }
+}
+
+// Pass all the data into specify fields
+   // Extract Fields
+   // Allow it to be specific to an array of objects
+   // Could add a function Process data and use specify/extract fields in it
+   // In process data do we want it allow the developer to choose what 
+
+
+export const fetchData = (url) => {
```

Comment:
Nice job, factoring this out!

*What is this functions single responsibility?*
The single responsibility is to fetch json data

*Question*
Can we rename this function to match it's single responsibility?